packages/contracts/src/contracts/current/protocol/AssetProxy/MixinERC20Transfer.sol

```diff
@@ -46,33 +46,64 @@ contract MixinERC20Transfer is
+        // | Header   | 0      | 4       | function selector                           |
+        // | Params   |        | 3 * 32  | function parameters:                        |
+        // |          | 4      |         |   1. from                                   |
+        // |          | 36     |         |   2. to                                     |
+        // |          | 68     |         |   3. amount                                 |
+
+        bytes4 transferFromSelector = IERC20Token(token).transferFrom.selector;
+        bool success;
         assembly {
-            if returndatasize {
-                success := 0
-                if eq(returndatasize, 32) {
-                    // First 64 bytes of memory are reserved scratch space
-                    returndatacopy(0, 0, 32)
-                    success := mload(0)
+            /////// Setup State ///////
+            // `cdStart` is the start of the calldata for `token.transferFrom` (equal to free memory ptr).
+            let cdStart := mload(64)
+
+            /////// Setup Header Area ///////
+            // This area holds the 4-byte `transferFromSelector`.
+            mstore(cdStart, transferFromSelector)
+            
+            /////// Setup Params Area ///////
+            // Each parameter is padded to 32-bytes. The entire Params Area is 96 bytes.
+            // A 20-byte mask is applied to addresses to zero-out the unused bytes.
+            mstore(add(cdStart, 4), and(from, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(add(cdStart, 36), and(to, 0xffffffffffffffffffffffffffffffffffffffff))
+            mstore(add(cdStart, 68), amount)
+
+            /////// Call `token.transferFrom` using the constructed calldata ///////
+            success := call(
+                gas,            // forward all gas
+                token,          // call address of token contract
+                0,              // don't send any ETH
+                cdStart,        // pointer to start of input
+                100,            // length of input
+                cdStart,        // write output over input
+                32              // output size is 32 bytes
+            )
+        
+            if success {
+                // Check return data.
+                // If there is no return data, we assume the token incorrectly
+                // does not return a bool. In this case we expect it to revert
+                // on failure, which was handled above.
+                // If the token does return data, we require that it is a single
+                // value that evaluates to true.
+                if returndatasize {
+                    success := 0
+                    if eq(returndatasize, 32) {
+                        // First 64 bytes of memory are reserved scratch space
+                        returndatacopy(0, 0, 32)
+                        success := mload(0)
```

Comment:
Since we ask for `call` to store the return data at `cdStart` (up to 32 bytes). We don't have to use `returndatacopy` again. Instead we can just use `success := mload(cdStart)`.